### DISCUSSION TITLE
Missing of meta data and fragments file

### DISCUSSION THREAD

**Discussion by mano2991** [stuart-lab/signac#1499] [Q&A]
Hello ,
I'm new to scATAC-seq data analysis. While reviewing various papers, I observed that Signac is frequently used for scATAC-seq analysis. I found a dataset on the GEO database with accession number [GSE153163,](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE153163) which corresponds to this article: [link](https://www.nature.com/articles/s41586-021-03670-5#data-availability). I'm attempting to replicate the same analysis as described in the paper. However, the authors did not provide the metadata and fragment files. Is it possible to analyze the data with Signac without these files? If so, could you guide me through the steps? Additionally, is there a way to extract these files from the provided h5 files?

I'm posting this on GitHub to seek further assistance in analyzing the data. Any guidance would be greatly appreciated.

**Comment by timoast** [comment]
Yes, you can run some steps without the fragment file. Whether it's sufficient depends on what your goal is, if you just want to cluster cells and find differential peaks then you can do that without a fragment file.

You can't create a fragment file from the h5 file

**Reply by mano2991** [reply]
Thanks, but I was trying to extract the fragment files from H5 files.  People might give your suggestion on this. If this is correct or not.  Alo @timoast can you give me the steps for it? Because i was not able to cluster my cell and find the peaks 

```
#required library 
library(rhdf5)
library(parallel)

#Define file paths
input_h5_path <- "ATAC_E18.h5"
output_tsv_path <- "E18.tsv"

# Read data from the .h5 file
barcodes <- h5read(input_h5_path, "matrix/barcodes")
features <- h5read(input_h5_path, "matrix/features/name")
data <- h5read(input_h5_path, "matrix/data")
indices <- h5read(input_h5_path, "matrix/indices")
indptr <- h5read(input_h5_path, "matrix/indptr")

# Process features to get chromosome, start, and end positions
features_data <- strsplit(features, ":")
chromosomes <- sapply(features_data, "[[", 1)
regions <- sapply(features_data, "[[", 2)
starts <- as.integer(sapply(strsplit(regions, "-"), "[[", 1))
ends <- as.integer(sapply(strsplit(regions, "-"), "[[", 2))

# Create the TSV file
con <- file(output_tsv_path, "w")
cat("Chromosome\tStart\tEnd\tBarcode\tCount\n", file=con)

for (barcode_idx in 1:(length(indptr) - 1)) {
  barcode <- barcodes[barcode_idx]
  
  # Correct the indexing for indptr to avoid negative subscripts
  start_idx <- indptr[barcode_idx] + 1
  end_idx <- indptr[barcode_idx + 1]
  
  for (idx in start_idx:end_idx) {
    feature_idx <- indices[idx]
    chrom <- chromosomes[feature_idx]
    start <- starts[feature_idx]
    end <- ends[feature_idx]
    count <- data[idx]
    cat(sprintf("%s\t%d\t%d\t%s\t%d\n", chrom, start, end, barcode, count), file=con)
  }
}

close(con)
```

**Reply by timoast** [reply]
The answer is you **cannot** create or extract a fragment file from the h5 file, they are different things

**Reply by mano2991** [reply]
@timoast Thanks. for your advice. I will run the pipeline without the fragment files
